src/SplitProvider.tsx

```diff
@@ -33,74 +25,75 @@ export const SplitContext: SplitReactContext = createContext<
  *    <App />
  *  </SplitProvider>
  */
-const SplitProvider = class extends React.Component<
-  ISplitProviderProps,
-  ISplitContextValues
-> {
-  constructor(props: ISplitProviderProps) {
-    super(props);
+const SplitProvider = ({ config, children }: ISplitProviderProps) => {
+  const {
+    startup = {},
```

Comment:
The default value of `startup` is an empty object, why? Will there be a case that `startup` will be a null and it needs to be set to `{}`? If that's the case, you're trying to get different values out of it below, this will potentially throw an exception when `startup` be just `{}`

Reply:
Correct the goal is just to get the values out of it below. It is an optional property so if it's omitted we default it to object here. null isn't valid and TS (if used) would catch that. As long as it is an object either passed in as one or default here, we can pull properties (whose values may be undefined - that's ok) without exceptions.